### ISSUE TITLE
Brotli prioritisation doesn't seem to work

### ISSUE THREAD

**Issue by mderriey** [shibayan/IIS.Compression.SiteExtension#23]
Hey 👋

Thanks a lot for the extension, we've had to remove Cloudflare in front of our app because some requests were taking too long and Cloudflare cut them off.
Since it was taking care of response compression, it's been super good to be able to have a no-code-changes solution that we could apply 🙏

I can see in the `.xdt` file that there's an IIS rewrite rule to prioritise using Brotli if it's present in the `Accept-Encoding` request header.
What we see, however, is that this prioritisation is not functioning on our app, such as:

- a request with `Accept-Encoding: gzip, deflate, br` returns a response with `Content-Encoding: gzip`; and
- a request with `Accept-Encoding: br` returns a response with `Content-Encoding: br`.

I'm not sure exactly if something within IIS could override this preference, but here's some information about the specific request:

- it's a `GET` request with no query string;
- it returns a JSON response, with the content type `application/json; charset=utf-8`;
- the uncompressed response is ~16MB.

I'm not sure what else could potentially explain this, so please let me know if I can provide any further information.
Thanks again!

**Comment by shibayan** [comment]
Try restarting the Web App and please tell me which application framework (ASP.NET / ASP.NET Core / Node.js / etc) you are using.

**Comment by mderriey** [comment]
Thanks for your prompt reply 👍

We restarted the app several times since the extension was installed, but the behaviour is the same.
It's an ASP.NET Core 3.0 app.

Thanks again.

**Comment by shibayan** [comment]
Thank you for answering. Two additional questions.

Are you using your own URL Rewrite Rule in your application's web.config?
And is Site Extensions installed elsewhere? For example, if you execute `clear` for URL Rewrite `rules` in web.config, it will not work.

If the configuration is correct, open Advanced Tools and use the Debug Console to check `D:\local\Config\applicationHost.config`.

![image](https://user-images.githubusercontent.com/1356444/72705589-3435c780-3b9f-11ea-9031-b9d7785f404f.png)

**Comment by mderriey** [comment]
Hey @shibayan 👋

We do use custom rewrite rules in the `web.config`, and we also have a `<clear />` element in the `web.config`.
We checked the `applicationHost.config` and the XDT transformation ran well as we can see the "Prioritize Brotli" rewrite rule.

Apologies for not thinking about this earlier, as the `<clear />` element in the `web.config` effectively ignores the rule in the `applicationHost.config`. We removed it now and can confirm that Brotli is used whenever the `Accept-Encoding` header contains `br`, even if other compression schemes are also present in the header.

Thanks a lot for your prompt support and the great extension 🙇‍♂️
